codec/src/test/java/io/netty/handler/codec/compression/Lz4FrameEncoderTest.java

```diff
@@ -52,4 +82,128 @@ protected ByteBuf decompress(ByteBuf compressed, int originalLength) throws Exce
 
         return Unpooled.wrappedBuffer(decompressed);
     }
+
+    @Test
+    public void testAllocateDirectBuffer() {
+        final int blockSize = 100;
+        testAllocateBuffer(blockSize, blockSize - 13, true);
+        testAllocateBuffer(blockSize, blockSize * 5, true);
+        testAllocateBuffer(blockSize, NONALLOCATABLE_SIZE, true);
+    }
+
+    @Test
+    public void testAllocateHeapBuffer() {
+        final int blockSize = 100;
+        testAllocateBuffer(blockSize, blockSize - 13, false);
+        testAllocateBuffer(blockSize, blockSize * 5, false);
+        testAllocateBuffer(blockSize, NONALLOCATABLE_SIZE, false);
+    }
+
+    private void testAllocateBuffer(int blockSize, int bufSize, boolean isDirect) {
+        // allocate the input buffer to an arbitrary size less than the blockSize
+        ByteBuf in = ByteBufAllocator.DEFAULT.buffer(bufSize, bufSize);
+        in.writerIndex(in.capacity());
+
+        ByteBuf out = null;
+        try {
+            Lz4FrameEncoder encoder = newEncoder(blockSize, Lz4FrameEncoder.DEFAULT_MAX_ENCODE_SIZE);
+            out = encoder.allocateBuffer(ctx, in, isDirect);
+            Assert.assertNotNull(out);
+            if (NONALLOCATABLE_SIZE == bufSize) {
+                Assert.assertFalse(out.isWritable());
+            } else {
+                Assert.assertTrue(out.writableBytes() > 0);
+                Assert.assertEquals(isDirect, out.isDirect());
+            }
+        } finally {
+            in.release();
+            if (out != null) {
+                out.release();
+            }
+        }
+    }
+
+    @Test (expected = EncoderException.class)
+    public void testAllocateDirectBuffer_ExceedMaxEncodeSize() {
```

Comment:
remove the `_` in the method name? we typically don't include this in method names.